lib/base/utility.cpp

```diff
@@ -1533,28 +1535,58 @@ static String UnameHelper(char type)
 	}
 }
 #endif /* _WIN32 */
+
+#ifdef _WIN32
+struct RegistryString
+{
+	union {
+		BYTE AsBytes[512];
+		char AsChars[1] = {0};
+	} Data;
```

Comment:
It would be nice to have documentation on what this type is used for. I was struggling to wrap my head around this until I found the term `type punning` about union types :).